code/modules/mod/mod_control.dm

```diff
@@ -444,9 +444,18 @@
 	icon_state = "[skin]-control[active ? "-sealed" : ""]"
 	return ..()
 
-/obj/item/mod/control/proc/set_wearer(mob/user)
+/// Called when an equipped MOD suit toggles activation.
+/obj/item/mod/control/proc/mod_suit_toggled(datum/source, mob/user)
+	SIGNAL_HANDLER
+	// The MOD helmet becomes unsealed if it's deactivated.
+	// Close internal air tank if MOD helmet was the only breathing apparatus.
+	if (wearer?.invalid_internals())
+		wearer.cutoff_internals()
+
+/obj/item/mod/control/proc/set_wearer(mob/living/carbon/human/user)
 	wearer = user
 	SEND_SIGNAL(src, COMSIG_MOD_WEARER_SET, wearer)
+	RegisterSignal(src, COMSIG_MOD_TOGGLED, .proc/mod_suit_toggled)
```

Comment:
What's the point of registering signal to your own proc, just call it in the proc.

Reply:
It's because I was advised to by some other maintainers. I can change it if you prefer

Reply:
Fixed by commit 40e719a5d56e10bc5f559e8aee02340feeb574fe